assets/user-profile/components/PauseNotificationModal.tsx

```diff
@@ -0,0 +1,152 @@
+        this.updateDate = this.updateDate.bind(this);
+        this.formRef = React.createRef<HTMLFormElement>();
+    }
+
+    componentDidMount() {
+        this.props.modalFormValid();
+    }
+
+    onSubmitForm(event: React.FormEvent<HTMLFormElement>) {
+        event.stopPropagation();
+        event.preventDefault();
+
+        if (this.formRef.current == null) {
+            throw new Error('ref missing');
+        }
+
+        const inputs = this.formRef.current.querySelectorAll('input') as NodeListOf<HTMLInputElement>;
+
+        const hasErrors = [...inputs].some(input => {
+            return input.checkValidity() !== true;
+        });
+
+        if (hasErrors === true) {
+            this.formRef.current.reportValidity();
+        } else {
+            this.props.updateUserNotificationPause(this.state);
+        }
+    }
+
+    updateDate(event: any, state: 'pauseFrom' | 'pauseTo') {
+        const stateClone: IState = {};
+
+        stateClone[state] = event?.target.value;
+        this.setState(stateClone);
+    }
+
+    disabledMinOption(state: string | undefined) {
+        if (state != '' && state != undefined) {
+            const newMaxDate = new Date(state);
+            newMaxDate.setDate(newMaxDate.getDate() + 1);
+            return newMaxDate.toISOString().split('T')[0];
+        } else {
+            return disabledOptionDateFrom;
+        }
+    }
+
+    disabledMaxOption(state: string | undefined) {
+        if (state != '' && state != undefined) {
+            const newMaxDate = new Date(state);
+            newMaxDate.setDate(newMaxDate.getDate() - 1);
+            return newMaxDate.toISOString().split('T')[0];
+        }
+    }
+
+    render() {
+        return (
+            <Modal
+                onSubmit={() => {
+                    this.formRef.current?.dispatchEvent(new Event('submit', {cancelable: true, bubbles: true}));
+                }}
+                title={gettext('Pause notifications...')}
+                onSubmitLabel={gettext('Pause notifications')}
+                disableButtonOnSubmit={false}
+                className="edit-schedule__modal"
+            >
+                <div className="nh-container nh-container--highlight rounded--none">
+                    <form
+                        ref={this.formRef}
+                        onSubmit={(event) => this.onSubmitForm(event)}
+                    >
+                        <div className='d-flex align-items-center gap-5'>
+                            <div className="d-flex flex-column align-items-start">
+                                <label htmlFor="date-from">{gettext('From')}</label>
+                                <input
+                                    id="date-from"
+                                    type="date"
+                                    name="date-from"
+                                    className="form-control"
+                                    onChange={(event) => this.updateDate(event, 'pauseFrom')}
```

Comment:
looks like we're expecting a return from this function which isn't the case, can you add curly braces to it?